core/src/main/java/com/linecorp/armeria/common/util/Sampler.java

```diff
@@ -80,7 +53,45 @@ public boolean isSampled() {
      *
      * @param rate minimum sample rate is 0.01, or 1% of traces
      */
-    public static Sampler create(float rate) {
-        return CountingSampler.create(rate);
+    static <T> Sampler<T> random(double rate) {
+        @SuppressWarnings("unchecked")
+        final Sampler<T> cast = CountingSampler.create(rate);
+        return cast;
+    }
+
+    /**
+     * Returns a sampler, given a rate-limited on a per-second interval.
+     *
+     * @param samplesPerSecond minimum rate-limited is 0 and the max is 2,147,483,647 (max int)
```

Comment:
How about:

    @param samplesPerSecond an integer between {@code 0} and {@value Integer#MAX_VALUE}

Reply:
Thank you for review. fiexed in 6459312.